metron-platform/metron-parsers/src/test/java/org/apache/metron/parsers/regex/RegularExpressionsParserTest.java

```diff
@@ -0,0 +1,152 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.metron.parsers.regex;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertTrue;
+
+public class RegularExpressionsParserTest {
+
+  private RegularExpressionsParser regularExpressionsParser;
+  private JSONObject parserConfig;
+
+  @Before
+  public void setUp() throws Exception {
+    regularExpressionsParser = new RegularExpressionsParser();
+  }
+
+  @Test
+  public void testSSHDParse() throws Exception {
+    String message =
+        "<38>Jun 20 15:01:17 deviceName sshd[11672]: Accepted publickey for prod from 22.22.22.22 port 55555 ssh2";
+
+    parserConfig = getJsonConfig(
+        Paths.get("src/test/resources/config/RegularExpressionsParserConfig.json").toString());
+    regularExpressionsParser.configure(parserConfig);
+    JSONObject parsed = parse(message);
+    // Expected
+    Map<String, Object> expectedJson = new HashMap<>();
+    expectedJson.put("device_name", "deviceName");
+    expectedJson.put("dst_process_name", "sshd");
+    expectedJson.put("dst_process_id", "11672");
+    expectedJson.put("dst_user_id", "prod");
+    expectedJson.put("ip_src_addr", "22.22.22.22");
+    expectedJson.put("ip_src_port", "55555");
+    expectedJson.put("app_protocol", "ssh2");
+    assertTrue(validate(expectedJson, parsed));
+
+  }
+
+  @Test
+  public void testNoMessageHeaderRegex() throws Exception {
+    String message =
+        "<38>Jun 20 15:01:17 deviceName sshd[11672]: Accepted publickey for prod from 22.22.22.22 port 55555 ssh2";
+    parserConfig = getJsonConfig(
+        Paths.get("src/test/resources/config/RegularExpressionsNoMessageHeaderParserConfig.json")
+            .toString());
+    regularExpressionsParser.configure(parserConfig);
+    JSONObject parsed = parse(message);
+    // Expected
+    Map<String, Object> expectedJson = new HashMap<>();
+    expectedJson.put("dst_process_name", "sshd");
+    expectedJson.put("dst_process_id", "11672");
+    expectedJson.put("dst_user_id", "prod");
+    expectedJson.put("ip_src_addr", "22.22.22.22");
+    expectedJson.put("ip_src_port", "55555");
+    expectedJson.put("app_protocol", "ssh2");
+    assertTrue(validate(expectedJson, parsed));
```

Comment:
I personally found junit logging to be insufficient. I wanted more information in the logs. Also expectedJson.put("ip_src_port", "55555"); was more concise than its counterpart.  

Other advantage of using this method was it would let you know all the failed scenarios in one run. While a failed JUnit assertion will stop the test case then and there itself. 

Also, Junit best practices state that maximum one assertion per test case. Now if we want to follow this best practice, we will have to write a unit test per field which again does not feel right. Having the validate method let us follow the Junit best practices.

Do you still want me to remove validate method ?

Reply:
> Junit best practices state that maximum one assertion per test case. 

I have never heard that, nor ever, ever followed that. :)  I think every test in Metron has multiple assertions, which are necessary.  

I think best practice may be to test one "thing" at a time, but you may require multiple assertions when testing that one "thing".

I think it is much simpler the way I suggested, but we could probably spend the time on other more important things.

Reply:
@nickwallen, I have changed the unit tests as you suggested.